rdr_service/tools/study_nph_biobank_file_export.py

```diff
@@ -0,0 +1,222 @@
+    ordered_sample_dao = NphOrderedSampleDao()
+    with ordered_sample_dao.session() as session:
+        return session.query(OrderedSample).filter(
+            OrderedSample.order_id == order_id
+        ).all()
+
+
+def _convert_ordered_samples_to_samples(order_id: int, ordered_samples: List[OrderedSample]) -> List[Dict[str, Any]]:
+    samples = []
+    for ordered_sample in ordered_samples:
+        supplemental_fields = ordered_sample.supplemental_fields if ordered_sample.supplemental_fields else {}
+        notes = ", ".join([f"{key}: {value}" for key, value in supplemental_fields.items()])
+        extract_volume_units = findall("[a-zA-Z]+", ordered_sample.volume)
+        volume_units = extract_volume_units[-1] if extract_volume_units else ""
+        sample = {
+            "sampleID": ordered_sample.aliquot_id,
+            "specimenCode": ordered_sample.identifier,
+            "kitID": order_id if ordered_sample.identifier.startswith("ST") else "",
+            "volume": ordered_sample.volume,
+            "volumeUOM": volume_units,
+            "processingDateUTC": ordered_sample.finalized,
+            "notes": notes,
+        }
+        samples.append(sample)
+    return samples
+
+
+def _get_rdr_participant_summary_for_nph_partipant(nph_participant_id: int) -> Optional[RdrParticipantSummary]:
+    nph_participant_dao = NphParticipantDao()
+    with nph_participant_dao.session() as nph_participant_session:
+        nph_participant: NphParticipant = nph_participant_session.query(NphParticipant).get(nph_participant_id)
+
+    rex_participant_mapping_dao = RexParticipantMappingDao()
+    with rex_participant_mapping_dao.session() as rex_sm_session:
+        rex_participant_mapping: RexParticipantMapping = (
+            rex_sm_session.query(RexParticipantMapping).filter(
+                RexParticipantMapping.ancillary_participant_id == nph_participant.id
+            ).first()
+        )
+
+    rdr_participant_id = rex_participant_mapping.primary_participant_id
+    rdr_participant_summary_dao = RdrParticipantSummaryDao()
+    with rdr_participant_summary_dao.session() as rdr_ps_session:
+        return rdr_ps_session.query(RdrParticipantSummary).filter(
+            RdrParticipantSummary.participantId == rdr_participant_id
+        ).first()
+
+
+def _convert_orders_to_collections(
+    orders: List[Order],
+    rdr_participant_summary: RdrParticipantSummary
+) -> List[Dict[str, Any]]:
+    collections = []
+    for order in orders:
+        samples = _convert_ordered_samples_to_samples(
+            order_id=order.id,
+            ordered_samples=_get_ordered_samples(order_id=order.id)
+        )
+        parent_study_category = _get_parent_study_category(order.category_id)
+        collections.append({
+            "visitID": parent_study_category.name if parent_study_category else "",
+            "timepointID": _get_study_category(order.category_id).name,
+            "orderID": order.id,
```

Comment:
`order.id` is our internal id value for this order. We need to be sending order id value we receive from Hpro on the api (`order.nph_order_id`).